src/timeline/InputBar.cpp

```diff
@@ -937,13 +940,34 @@ InputBar::command(const QString &command, QString args)
                                 cache::getMembers(this->room->roomId().toStdString(), 0, -1));
     } else if (command == QLatin1String("converttoroom")) {
         utils::removeDirectFromRoom(this->room->roomId());
+    } else if (command == QLatin1String("ignore")) {
+        this->setArgIgnored(args, true);
+    } else if (command == QLatin1String("unignore")) {
+        this->setArgIgnored(args, false);
     } else {
         return false;
     }
 
     return true;
 }
 
+void
+InputBar::setArgIgnored(const QString &user, const bool ignored)
+{
+    QSharedPointer<UserProfile> profile(
+      new UserProfile(QString(), user, TimelineViewManager::instance()));
+    connect(profile.get(), &UserProfile::failedToFetchProfile, [user] {
+        MainWindow::instance()->showNotification(tr("Failed to fetch user %1").arg(user));
+    });
+
+    connect(profile.get(),
+            &UserProfile::globalUsernameRetrieved,
+            [profile, ignored](const QString &user_id) {
+                Q_UNUSED(user_id)
+                profile->setIgnored(ignored);
+            });
```

Comment:
This is a memory leak, since you never let go of the shared pointer. I suggest you instead allocate raw and then do a deleteLater on the captured pointer in each of the signal handlers. Afaik there is no better way to do that rn, unless you don't wait for the profile to fetch.

Reply:
Ah right, I imagine since the connection would only be deleted (thus destroying the lambda) once the profile was destroyed, which never happens since that would require the lambda being destroyed to begin with. Going to fix it.